release_process.md

```diff
@@ -1,162 +1,550 @@
-# Release process
+# Release Process
+
+Lance uses a semantic versioning release workflow with Release Candidates (RC) for voting before stable releases. The process features linear commit history on branches.
+
+## Overview
+
+The release process follows these stages:
+1. **Development** on `main` branch with beta versions
+2. **Preview Releases** published from beta versions (beta.1+)
+3. **Release Candidates** created for voting (rc.1, rc.2, etc.)
+4. **Stable Releases** promoted from approved RCs
+5. **Patch Releases** managed through release branches
```

Comment:
It's not clear here (and maybe it doesn't need to be) if the release branches are introduced at the release candidates stage or the patch releases stage.

Reply:
superseded by https://github.com/lancedb/lance/pull/5089#discussion_r2495914866 which mentions the release branch